File: neetcode-150/java/Two-Pointers/maxArea.java
class Solution {
    public int maxArea(int[] heights) {
        int left = 0, right = heights.length - 1, res = 0;
        while (left < right) {
            int length = right - left;
            int area = length * Math.min(heights[left], heights[right]);
            res = Math.max(res, area);
            if (heights[left] < heights[right]) {
                left++;
            } else {
                right--;
            }
        }
        return res;
    }
}

File: nexon/01-04-2025/maxArea.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l, r = 0, len(heights) - 1
        res = 0
        while l < r:
            area = (r - l)*min(heights[l], heights[r])
            if (heights[l] < heights[r]):
                l += 1
            else:
                r -= 1
            res = max(res, area)
        return res
        